### DISCUSSION TITLE
Single and Double Click on p-table

### DISCUSSION THREAD

**Discussion by Hamza5344** [primefaces/community#66] [PrimeNG] [answered]
I am currently having a problem in p-table. i am trying to add two funktions on a table. i want to triger one funktion on single click on table row and another funktion on double click on the Table row. by single and double click i am only able to trigger the first function.
i am using the following two functions: 
(click)="onRowSelect($event, 'single')" -> should trigger on single Click
(dblclick)="onRowDblclick($event, 'double')" -> shoulfd trigger on double Click
  
  
  
  onRowSelect(event: any, eventType: string) {
    console.log('event', event);
    console.log('eventType', eventType);
    }
    
    onRowDblclick(event: Event, eventType: string) {
    console.log('Doppelklick auf Zeile:', event);
    console.log('eventType', eventType);
  }

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,
In Angular, differentiating between single-click and double-click events can be a bit tricky because both events start with a single click. This means that when a double-click event happens, a single-click event also fires immediately before it.

One way to handle this is to introduce a slight delay on the single-click event to check if a subsequent click happens, signaling a double-click.

Here’s a basic example of how you can differentiate between single-click and double-click events in any component:

```
<div 
    (click)="handleSingleClick()" 
    (dblclick)="handleDoubleClick()">
    Click
</div>

```
```
import { Component } from '@angular/core';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css']
})
export class AppComponent {
  singleClickTimer: any; // Used for the setTimeout
  delay: number = 250; // Delay in milliseconds

  handleSingleClick() {
    // Clear any existing single click timer to ensure 
    // we don't run multiple timers at the same time
    clearTimeout(this.singleClickTimer);

    // Set a new timer
    this.singleClickTimer = setTimeout(() => {
      console.log('Single Click');
    }, this.delay);
  }

  handleDoubleClick() {
    // Clear the single click timer to prevent it from executing
    clearTimeout(this.singleClickTimer);
    console.log('Double Click');
  }
}
```

The logic works as follows:

- When a click event occurs, the **handleSingleClick** method is called. This method sets a timer that, after a short delay, will print "Single Click".
- If a double-click event occurs (within the delay period), the handleDoubleClick method is called which will clear the timer for the single click (to prevent it from executing) and then print "Double Click".
Adjust the **delay** value if you need a longer or shorter wait time to differentiate between single and double clicks.

Keep in mind that this approach relies on JavaScript's setTimeout and clearTimeout functions, which means there's a slight delay in processing single-click events. This is usually not noticeable to users, but it's something to be aware of if precise timing is crucial for your application.

[Here is a Stackblitz example of how to do it](https://stackblitz.com/edit/8n4zfy?file=src%2Fapp%2Fdemo%2Ftoast-basic-demo.html,src%2Fapp%2Fdemo%2Ftoast-basic-demo.ts)

**Reply by Hamza5344** [reply]
yes its good but all of the work is being handle manually. Right now i have to give a time manually. is there any default Method in angular for that ?

**Reply by cetincakiroglu** [reply]
Unfortunately, there isn't a default method for that, you need to handle it custom
